### DISCUSSION TITLE
nvm: command not found

### DISCUSSION THREAD

**Discussion by Lidbetter** [community/community#25344] [Actions]
<p>Hello,</p>
<p>I am migrating CI from travis to actions and I am having problems using nvm.<br>
looking at <a href="https://github.com/actions/virtual-environments/blob/master/images/linux/Ubuntu1804-README.md" rel="nofollow noopener">actions/virtual-environments Ubuntu1804-README</a> <br>
nvm should be available, but when I run my action it errors with:</p>
<pre><code>nvm: command not found
</code></pre>
<p>My action and script look like this:</p>
<p>workflow.yml:</p>
<pre><code>jobs:
  test:
    runs-on: ubuntu-latest

    steps:
      - name: Checkout
        uses: actions/checkout@v2

      - name: Build
        shell: bash -l -eo pipefail {0}
        run: .build.sh
</code></pre>
<p>.build.sh:</p>
<pre><code>#!/bin/bash

set -eo pipefail

# (re)source nvm if required
command -v nvm &gt;/dev/null 2&gt;&amp;1 || {
  if [[-s "${NVM_DIR}/nvm.sh"]]; then
    source "${NVM_DIR}/nvm.sh"
  elif [[-s "${HOME}/.nvm/nvm.sh"]]; then
    source "${HOME}/.nvm/nvm.sh"
  fi
}

nvm install 12.16.0
nvm use 12.16.0

yarn install --production=true --frozen-lockfile

yarn link
</code></pre>
<p>What am I doing wrong here, how do I use the image provided nvm in my script?</p>
<p>Edit:<br>
After looking at the output of the ‘setup job’ step, I see that my action is being run against version 20200323.1  I should have noticed this sooner.</p>
<p>The version (as of making this edit [16 April 2020]) is old, there have been 3 tagged newer releases.<br>
Is getting a version of the virtual envionment this far behind the current release expected?<br>
Am I garanteed (per repo?) to get a consistent version when my action is run or maybe atleast no version older than X (I can see this leading to flakey action outcomes if I sometimes get a version with features and sometimes without)</p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/lidbetter">@lidbetter</a> ,</p>
<p>I have reported this ticket to the appropriate engineering team for further investigation and evaluation. If they have any progress, I will notify you in time, and sometimes the appropriate engineers may directly reply you here.</p>

**Comment by maxim-lobanov** [comment]
<p>Hello,</p>
<p>We have confirmed the issue.</p>
<p>New image with NVM was not deployed to some VM machines / pools. Looking at it</p>

**Comment by Lidbetter** [comment]
<p>Thanks <a class="mention" href="/brightran">@brightran</a> </p>
<p>I have updated my original post with some more details.<br>
My action was being run on an older version of the virtual environment, which explains why it wasn’t working.</p>
<p>Is there anything I can do other than wait for updates to be rolled out - is there anywhere I can see progress and/or what possible versions my action could be run on</p>

**Comment by al-cheb** [comment]
<p>Hello,<br>
All pools have been updated. Could you please check if nvm works for you?</p>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/lidbetter">@lidbetter</a> ,</p>
<p>The engineering team has fixed this issue.</p>
<p>I checked with the following scripts, and I can confirm the  <strong>nvm</strong> command can work fine on the GitHub-hosted runners now.</p>
<pre><code>jobs:
  test:
    name: Test
    runs-on: ubuntu-latest
    steps:
      - name: check nvm location
        shell: bash -l -eo pipefail {0}
        run: echo $NVM_DIR

      - name: test nvm command
        shell: bash -l -eo pipefail {0}
        run: nvm ls
</code></pre>
<p>Could you please also check this on your side? Any progress, feel free to tell us.</p>

**Comment by Lidbetter** [comment]
<p>Thanks, nvm is now working!</p>

**Comment by Lidbetter** [comment]
<p>Thanks, nvm is now working for me</p>

**Comment by chingc** [comment]
<p>Gh-hosted runners have nvm now?  Thank you!</p>
<p>I can finally get rid of the <code>set-output</code> workaround to read my .nvmrc with setup-node.</p>

**Comment by rachelcarmena** [comment]
<p>Hi!</p>
<p>Please, is it still working for you? It’s available according to the list of sw: <a href="https://github.com/actions/virtual-environments/blob/ubuntu18/20200604.1/images/linux/Ubuntu1804-README.md" rel="nofollow noopener">https://github.com/actions/virtual-environments/blob/ubuntu18/20200604.1/images/linux/Ubuntu1804-README.md</a></p>
<p>However, I’m getting <code>nvm: command not found</code> with <code>ubuntu-latest</code> right now.</p>
<p>I looked at the HOME directory and <code>.nvm</code> is available so it’s possible to finish the installation:</p>
<pre><code class="lang-auto">  export NVM_DIR="$HOME/.nvm"
  source "$NVM_DIR/nvm.sh"
</code></pre>
<p>However it doesn’t recognize all the available Node versions according to the list of sw:</p>
<pre><code class="lang-auto">node:

    node 8.17.0
    node 10.21.0
    node 12.18.0
    node 14.4.0
</code></pre>
<p>Thanks in advance!</p>
